File: DentRec.Domain/Entities/Patient.cs
namespace DentRec.Domain.Entities
{
    public class Patient : BaseEntity
    {
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Gender { get; set; } = string.Empty;
        public int Age { get; set; } // Use age because birthday is not required
        public string? Phone { get; set; }
        public string? Email { get; set; }
        public string? Address { get; set; }

        public ICollection<PatientLog> PatientLogs { get; set; } = new List<PatientLog>();
        public ICollection<PatientPrescription> PatientPrescriptions { get; set; } = new List<PatientPrescription>();
        public ICollection<Payment> Payments { get; set; } = new List<Payment>();
        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    }
}

File: Client/src/app/shared/models/patient.ts
import { PatientLog } from "./patientLog"
import { Payment } from "./payment"

export interface Patient {
    id: number
    firstName: string
    lastName: string
    gender: string
    age: number
    phone: string
    email: string
    address: string
  }

  export interface PatientDetails {
    id: number
    firstName: string
    lastName: string
    gender: string
    age: number
    phone: string
    email: string
    address: string
    payments: Payment[]
    patientLogs: PatientLog[]
    createdOn: string
    modifiedOn: string
    createdBy: string
    modifiedBy: string
  }